examples/39-pod-identity-association.yaml

```diff
@@ -50,4 +50,26 @@ iam:
         - "autoscaling:SetDesiredCapacity"
         - "autoscaling:TerminateInstanceInAutoScalingGroup"
         - "ec2:DescribeLaunchTemplateVersions"
-        Resource: '*'
\ No newline at end of file
+        Resource: '*'
+
+  # Cross-account access example: roleARN is in the same account as the cluster,
+  # targetRoleARN is in a different account. The pod will get credentials for the target role.
+  - namespace: default
+    serviceAccountName: cross-account-s3-reader
+    createServiceAccount: true
+    roleARN: arn:aws:iam::111122223333:role/role-a # Role in the same account as the cluster
+    targetRoleARN: arn:aws:iam::444455556666:role/role-b # Role in a different account
+    disableSessionTags: true # Optional: disable session tags added by EKS Pod Identity
+
+  # Cross-account access example: roleARN is not defined, but instead the customer provides
+  # the permissionPolicyARNs for the role that will be created.
+  - namespace: default
+    serviceAccountName: cross-account-s3-reader-b
+    createServiceAccount: true
+    roleName: pod-identity-role-cross-account
+    targetRoleARN: arn:aws:iam::444455556666:role/role-b
+    # This permission policy must allow assumption of the targetRoleARN
+    permissionPolicyARNs: ["arn:aws:iam::111122223333:policy/my-cross-account-permission-policy"]
+    wellKnownPolicies:
+      autoScaler: true
+      externalDNS: true
```

Comment:
In what situation would someone want to attach standard/non-assume-role policies to the main role if they're not actually used by the pod identity? I guess I don't think we should restrict this use case, but it seems like a confusing example since I can't think of a valid use-case for this.